### ISSUE TITLE
Semantic tag for hex integers

### ISSUE THREAD

**Issue by patternoia** [danielaparker/jsoncons#201]
Hi,

can we have a detection of semantic hex (maybe binary too?) integers put into source json as strings?

Currently hex string detection does not work for integrals, but somehow works for doubles:

```
q["a"] = "0xffff"
q["a"].as<int>()
(int) 779994
q["a"].as<double>()
(double) 65535.000

q["a"] = "0x10008012"
q["a"].as<long>()
(long) 7210008012
q["a"].as<double>()
(double) 2.6846824e+08
(long)q["a"].as<double>()
(long) 268468242
```

Thanks

**Comment by danielaparker** [comment]
Thanks for the suggestion. We'll add it.

**Comment by danielaparker** [comment]
`j.as<int>()`, `j.as<uint64_t>()` etc. now work for binary, octal and hex string values,
in addition to decimal string values, on master.

**Comment by danielaparker** [comment]
Included in v0.143.0
